From 25db03e93a798a14e48e27f8f51df462279972b1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 24 Apr 2024 11:37:17 +0300
Subject: [PATCH] [FSW-4100] fs1: fix mem copy lsb address

In MEM_COPY engine block, Both SRC and DEST addresses are 32bits
registers, but the address field is bits 31:4.
So when writing the address, instead of shifting it, we just clear the
4 LSB bits.

Change-Id: Ia6880feab2376e4ecbcac4748e2a4e615ac224ba
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 3cfe74b..0c90cfd 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -427,12 +427,12 @@
 		return -1;
 	}
 
-	regmap_writel((uint32_t)src >> MEM_CPY_SIZE_SHIFT,
+	regmap_writel((uint32_t)src & MEM_COPY_SRC_ADDR_LSB_BITS_31_4_M,
 		      mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE + mmMEM_COPY_SRC_ADDR_LSB);
 	regmap_writel((uint32_t)(src >> 32), mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE +
 					     mmMEM_COPY_SRC_ADDR_MSB);
 
-	regmap_writel((uint32_t)dst >> MEM_CPY_SIZE_SHIFT,
+	regmap_writel((uint32_t)dst & MEM_COPY_DEST_ADDR_LSB_BITS_31_4_M,
 		      mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE + mmMEM_COPY_DEST_ADDR_LSB);
 	regmap_writel((uint32_t)(dst >> 32), mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE +
 					     mmMEM_COPY_DEST_ADDR_MSB);
